plugin/mlsdbprovider.cpp

```diff
@@ -134,51 +134,74 @@ MlsdbProvider::~MlsdbProvider()
         staticProvider = 0;
 }
 
-/* TODO: coalesce lookups to avoid unnecessary repeated file I/O */
-bool MlsdbProvider::searchForCellIdLocation(const MlsdbUniqueCellId &uniqueCellId, MlsdbCoords *coords)
-{
-    // try to find the mlsdb data file which should contain it.
-    // the mlsdb data files are separated into "first digit of location code" directories/buckets.
-    QChar firstDigitAreaCode = QString::number(uniqueCellId.locationCode()).at(0);
-    QDirIterator it("/usr/share/geoclue-provider-mlsdb/", QDirIterator::Subdirectories);
-    while (it.hasNext()) {
-        const QString fname(it.next());
-        if (fname.endsWith(QStringLiteral("/%1/mlsdb.data").arg(firstDigitAreaCode), Qt::CaseInsensitive)) {
-            // found an mlsdb.data file which might contain the cell data.  search it.
-            QFile file(fname);
-            file.open(QIODevice::ReadOnly);
-            QDataStream in(&file);
-            quint32 magic = 0, expectedMagic = (quint32)0xc710cdb;
-            in >> magic;
-            if (magic != 0xc710cdb) {
-                qCDebug(lcGeoclueMlsdb) << "geoclue-mlsdb data file" << fname << "format unknown:" << magic << "expected:" << expectedMagic;
-                continue; // ignore this file
-            }
-            qint32 version;
-            in >> version;
-            if (version != 3) {
-                qCDebug(lcGeoclueMlsdb) << "geoclue-mlsdb data file" << fname << "version unknown:" << version;
-                continue; // ignore this file
+// TODO: Search alternative locations for files with mlsdb data
+bool MlsdbProvider::searchForCellIdLocation(quint64 uniqueCellId, MlsdbCoords *coords) {
+    if (uniqueCellId == 0) {
+        return false;
+    }
+    QString df = "/usr/share/geoclue-provider-mlsdb/data/";
+    QString lf = "/usr/share/geoclue-provider-mlsdb/data/";
```

Comment:
Could prefer a bit more verbose variable naming. Couple character names are harder to reason about and to search their usage.